File: 프로그래머스/Lv.1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    temp = []
    for value in arr:
        if (len(temp) == 0) :
            temp.append(value)
        else :
            if (temp[-1] != value) :
                answer.append(temp[-1])
                temp.clear()
                temp.append(value)
    if(len(temp) != 0) :
        answer.append(temp[-1])
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stackInt = new Stack<>();
        List<Integer> answer = new ArrayList<>();
        
        for (Integer i = 0; i < arr.length; i++) {
            if (stackInt.isEmpty()) {
                stackInt.push(arr[i]);
            }
            
            Integer stackTop = stackInt.pop();
            if (stackTop != arr[i]) {
                answer.add(stackTop);
            }
            stackInt.push(arr[i]);
        }
        answer.add(stackInt.pop());

        return answer.stream().mapToInt(n->n).toArray();
    }
}